api/src/services/items.ts

```diff
@@ -175,18 +176,26 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
 			const payloadWithTypeCasting = await payloadService.processValues('create', payloadWithoutAliases);
 
 			// In case of manual string / UUID primary keys, the PK already exists in the object we're saving.
-			let primaryKey = payloadWithTypeCasting[primaryKeyField];
+			let primaryKey = payloadWithTypeCasting[primaryKeyFieldName];
 
 			try {
+				if (!payloadWithoutAliases[primaryKeyFieldName] && primaryKeyField?.defaultValue === 'AUTO_INCREMENT') {
```

Comment:
`AUTO_INCREMENT` seems to be constant across DB dialects

Reply:
Yes, the `@directus/schema` sets the default value as `AUTO_INCREMENT` accordingly. 👍 

### Reference

https://github.com/directus/directus/blob/5107c13f747e66dc79e653cfd85d7194326103a8/packages/schema/src/dialects/postgres.ts#L185-L187

https://github.com/directus/directus/blob/5107c13f747e66dc79e653cfd85d7194326103a8/packages/schema/src/dialects/sqlite.ts#L62-L65

https://github.com/directus/directus/blob/5107c13f747e66dc79e653cfd85d7194326103a8/packages/schema/src/dialects/mysql.ts#L127

https://github.com/directus/directus/blob/5107c13f747e66dc79e653cfd85d7194326103a8/packages/schema/src/dialects/cockroachdb.ts#L190-L192

https://github.com/directus/directus/blob/5107c13f747e66dc79e653cfd85d7194326103a8/packages/schema/src/dialects/mssql.ts#L161

https://github.com/directus/directus/blob/5107c13f747e66dc79e653cfd85d7194326103a8/packages/schema/src/dialects/oracledb.ts#L154